src/util/strutil.c

```diff
@@ -8,6 +8,19 @@
 #include <ctype.h>
 #include "RG.h"
 #include "rmalloc.h"
+#include "deps/utf8proc/utf8proc.h"
+
+int str_length(const char *str) {
+	size_t str_len = strlen(str);
+	utf8proc_int32_t c;
+	size_t i = 0;
+	int len = 0;
+	while(i < str_len) {
```

Comment:
I think we can do the following:
`str_len` is the length in bytes (should be rename to reflect that!)
`i = utf8proc_iterate(str + i, -1, &c);` gives us the number of bytes for a single character
`i / str_len` should be the length of the string in characters

Reply:
each character can be different size